lib/Common/Memory/PageAllocator.cpp

```diff
@@ -929,7 +929,25 @@ PageAllocatorBase<T>::FillFreePages(__in void * address, uint pageCount)
 #endif
     if (ZeroPages())
     {
-        memset(address, 0, AutoSystemInfo::PageSize * pageCount);
+        //
+        // Do memset via non-temporal store to avoid evicting existing processor cache.
+        // This helps low-end machines with limited cache size.
+        //
+#if defined(_M_IX86) || defined(_M_X64)
+        if (CONFIG_FLAG(ZeroMemoryWithNonTemporalStore))
+        {
+            size_t writeBytes = 0;
+            __m128i simdZero = _mm_setzero_si128();
+            for (__m128i * p = (__m128i *)address; writeBytes < pageCount * AutoSystemInfo::PageSize; p += 1, writeBytes += sizeof(__m128i))
+            {
+                _mm_stream_si128(p, simdZero);
+            }
+        }
+        else
+#endif
+        {
+            memset(address, 0, AutoSystemInfo::PageSize * pageCount);
+        }
```

Comment:
No rush to get this in.  Let’s find a good place for this function and check it in then.

From: Tom Tan [mailto:notifications@github.com]
Sent: Thursday, July 28, 2016 10:30 AM
To: Microsoft/ChakraCore <ChakraCore@noreply.github.com>
Cc: Louis Lafreniere <louisl@microsoft.com>; Mention <mention@noreply.github.com>
Subject: Re: [Microsoft/ChakraCore] Zero free memory with non-temporal stores (#1311)


In lib/Common/Memory/PageAllocator.cpp<https://github.com/Microsoft/ChakraCore/pull/1311#discussion_r72665328>:

> +        //

> +#if defined(_M_IX86) || defined(_M_X64)

> +        if (CONFIG_FLAG(ZeroMemoryWithNonTemporalStore))

> +        {

> +            size_t writeBytes = 0;

> +            __m128i simdZero = _mm_setzero_si128();

> +            for (__m128i * p = (__m128i *)address; writeBytes < pageCount * AutoSystemInfo::PageSize; p += 1, writeBytes += sizeof(__m128i))

> +            {

> +                _mm_stream_si128(p, simdZero);

> +            }

> +        }

> +        else

> +#endif

> +        {

> +            memset(address, 0, AutoSystemInfo::PageSize * pageCount);

> +        }

PageAllcoator.cpp doesn't like the ideal place to put general library functions. I suggest merge the current change and I will figure it out that why we get multiple definition error in app.cpp which is not supposed to happen.

________________________________

In reply to: 72664236<https://github.com/Microsoft/ChakraCore/pull/1311#discussion_r72664236>

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/Microsoft/ChakraCore/pull/1311/files/97e55d90dcec6537e90f0a47a6272a5d66516a3d#r72665328>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AL-Bt0nOHkxxtemurT1OIGwvFzfkUyAEks5qaOcngaJpZM4JSW9g>.